### DISCUSSION TITLE
Stray (unwanted) guests on Private servers?

### DISCUSSION THREAD

**Discussion by pgScorpio** [jamulussoftware/jamulus#2243] [Technical Support & Help]
I'm doing technical support for several Dutch Jamulus online events.
We use several **Private** servers, all on IP addresses never used for public Jamulus servers and all servers are on **non-standard port numbers**. Yet we see more and more often stray "guests" (mostly from the USA) appearing in our sessions...

Unfortunately most of the time we have no time to ask how they got here, but the few times we could ask they said that they selected a (random) server from Jamulus explorer.

**How is this possible????** Our servers do NOT appear on Jamulus explorer! (unless added as "Custom directory server", but than it should still be visible only by the ones who added the "Custom directory server" to their list right??)

**Comment by ann0see** [comment]
This sounds very strange. Can you post your config file (with all CLI arguments)?

Also you might use a firewall like worldjam. @chrisrimple

**Reply by pgScorpio** [reply]
> Can you post your config file (with all CLI arguments)?

Well that are a lot of different configs, but all are like:

`ExecStart=/bin/jamulus-headless -n -s -T \ 

-u 70 \ 

-p 500nn \ 

--directoryserver "xxx.yyy.nl" \ 

--serverinfo "roomname;TransIp Amsterdam;151" \ 

-w "/home/jamulus/welcome/roomname.msg.txt"
`

All servers are registering to a private server with similar config, but -p 22124 --directoryserver localhost
but AFAIK none of the stray guests entered via our custom directory servers.

**Reply by pljones** [reply]
Do any of the servers use the GUI interface rather than command line?

**Reply by pgScorpio** [reply]
> Do any of the servers use the GUI interface rather than command line?

No. All are on headless Ubuntu...

I strongly suspect a bug on Jamulus explorer, since afaik all stray clients entered via explorer and all servers it occured on had there (private) directory server entered as custom server on explorer by somebody at some point in time.
Also I hear other complaints from clients about occasionally entering the wrong server (also public servers) when selected via explorer.
Maybe some kind of caching problem??

**Reply by pljones** [reply]
You're saying it's sharing data between user sessions?

I'm not clear what entering via Explorer means, either.  How do you do that?

**Reply by pgScorpio** [reply]
Not familiar with Jamulus Explorer? [(explorer.jamulus.io)](https://explorer.jamulus.io/)
![Edit servver list](https://user-images.githubusercontent.com/9221087/149636169-7c2188c3-db14-4262-85bf-ba51d37e7f0f.png)



![Server view](https://user-images.githubusercontent.com/9221087/149635588-78f8bf27-42f9-4bcd-a1c8-f044d8e50792.png)

So I guess Jamulus explorer is either sometimes showing custom servers to others (mixed in with the public ones?), or it's somehow messing up IP:port addresses. I wouldn't how they otherwise can "enter via Jamulus explorer", since that's what at least two people said...

**Reply by ann0see** [reply]
@pgScorpio edited your comment since the server name was visible.

**Reply by pljones** [reply]
@pgScorpio Yes, I'm familiar with it.  The source code's available if you'd like to take a look through it.  What I said was that I didn't understand the term used - I've never heard it before and I wondered if you knew what was meant by it.  I've had people say they can see a server on Jamulus Explorer and used the details - but I've not heard _that_ termed "Entering via Jamulus Explorer", presumably because you're using Jamulus to actually connect.  Even the word "Enter" is a bit alien.

**Reply by pgScorpio** [reply]
Yes, I understand your confusion. I had the same, so I should have been more clear... But it's quite a story...

We had several times multiple strangers (first time over 10) appearing on our private servers of online events. Most of the time we just wanted to get rid of them ASAP, so no time to ask questions, but two times we did ask "How did you get here?" and both times they said "via Jamulus explorer", so apparently they must have used IP:port to connect, since there is no way to connect directly via explorer. No guarantee though that they all did connect this way, because, as said, most of the time we just told them to leave immediately without asking questions, but I don't see any connection between Jamulus explorer and the normal directory servers (except that explorer is querying them). But, after hearing the "via Jamulus explorer" I asked around if somebody in the crew had used our servers in Jamulus explorer. One did, and he immediately removed our directory server from his Jamulus explorer custom directory list. After that the number of stray "guests" diminished and after about an hour we had no more stray guests. The second time we had some stray guests and again a crew member had used Jamulus explorer to check our server IP addresses prior to the event. 
After that we had two events where we all explicitly did NOT use Jamulus explorer and those where without stray guests.  
Last week I used Jamulus explorer myself to check my servers  (not before an event fortunately) and again I had a stray guest on one of my servers again! And, since this time I had all the time, I tried to find out more, but he just didn't respond to voice or chat unfortunately. 
But it confirms to me that the problem really seems to be related to Jamulus explorer somehow. I just don't understand how.

**Reply by pljones** [reply]
Sounds like you might be best raising an issue with Tony, then: https://github.com/softins/jamulus-php/issues

**Reply by pgScorpio** [reply]
I guess [Jamulus-web](https://github.com/softins/jamulus-web) would be a better place to raise an issue then???

**Reply by pljones** [reply]
Depends if it's the PHP code or the HTML code.
